src/main/java/uk/gov/hmcts/ccd/domain/service/caseaccess/CaseAccessOperation.java

```diff
@@ -84,25 +92,118 @@ public void updateUserAccess(CaseDetails caseDetails, CaseUser caseUser) {
     }
 
     @Transactional
-    public void addCaseUserRoles(List<CaseAssignedUserRole> caseUserRoles) {
+    public void addCaseUserRoles(List<CaseAssignedUserRoleRequest> caseUserRoles) {
+
+        Map<Long, List<CaseAssignedUserRoleRequest>> cauRolesByCaseId = getMapOfCaseAssignedUserRolesByCaseId(caseUserRoles);
+
+        Map<String, Map<String, Object>> newUserCounts = getNewUserCountByCaseAndOrganisation(cauRolesByCaseId);
+
+        cauRolesByCaseId.forEach((caseId, roles) ->
+            roles.forEach(caseUserRole ->
+                caseUserRepository.grantAccess(caseId, caseUserRole.getUserId(), caseUserRole.getCaseRole()))
+        );
+
+        if (newUserCounts != null) {
```

Comment:
newUserCounts returned shouldn't be nullable. If there is nothing to return, the method calculating this should return an empty map.